src/views/OnBoardingSubProduct/components/SubProductStepSubmit.tsx

```diff
@@ -139,14 +143,19 @@ function SubProductStepSubmit({
         subproduct_id: subProduct.id,
       });
       forward();
-    } else if (outcome === 'error') {
+    } else if (
+      outcome === 'error' &&
+      (postLegalsResponse as AxiosError<Problem>).response?.status !== 409
+    ) {
       setError(true);
       trackEvent('ONBOARDING_SEND_FAILURE', {
         party_id: externalInstitutionId,
         request_id: requestId,
         product_id: product?.id,
         subproduct_id: subProduct?.id,
       });
+    } else {
+      setClientBadInputError(true);
```

Comment:
it's more scalable as solution to switch this if/else branch.

the else condition should be the more general case, while the specific case of 409 should be checked and handled (if in the future we will add other cases, this structure will be more difficult to adapt than the suggested